### PR TITLE
Fix: Remove unnecessary files and resolve class dependencies

### PR DESCRIPTION
**Pull request by Curodenz**
- remove the files in newer version but useless in 1.17.1
- resolve the dependency of the interface FloatUnaryOperator, I use the implementation in fastutil 8.12.1
- adapt to fabric api v0.41.0

### CODE REVIEW

File: src/main/resources/assets/jade/lang/README.txt
Comment (Snownee) [inline src/main/resources/assets/jade/lang/README.txt:1]: This file should only exist in 1.20+ branches

File: src/main/resources/fabric.mod.json [hunk 1]
```diff
@@ -5,17 +5,20 @@
...
     "homepage": "https://www.curseforge.com/minecraft/mc-mods/jade",
     "sources": "https://github.com/Snownee/Jade",
     "issues": "https://github.com/Snownee/Jade/issues"
   },
 
   "license": "CC BY-NC-SA 4.0",
+  "icon": "icon.png",
 
+  "environment": "*",
   "entrypoints": {
```
Comment (Snownee) [inline src/main/resources/fabric.mod.json:22]: No entrypoint declaration for REI

File: src/main/resources/fabric.mod.json [hunk 2]
```diff
@@ -30,25 +33,20 @@
...
   },
-
-  "custom": {
-    "modmenu": {
-      "links": {
-        "modmenu.discord": "https://discord.gg/KzGQW7a"
-      }
-    }
-  }
+  "suggests": {
```
Comment (Snownee) [inline src/main/resources/fabric.mod.json]: Remove it

File: src/main/resources/fabric.mod.json [hunk 3]
```diff
@@ -30,25 +33,20 @@
...
+    "fabricloader": ">=0.11.3",
+    "fabric": ">=0.41.0",
+    "minecraft": "1.17.x",
+    "java": ">=16"
   },
-
-  "custom": {
-    "modmenu": {
-      "links": {
-        "modmenu.discord": "https://discord.gg/KzGQW7a"
```
Comment (Snownee) [inline src/main/resources/fabric.mod.json:50]: This should not be removed

File: src/main/java/snownee/jade/addon/vanilla/FurnaceProvider.java [hunk 4]
```diff
@@ -51,7 +51,11 @@ public void appendServerData(CompoundTag data, ServerPlayer player, Level world,
 			items.add(furnace.getItem(i).save(new CompoundTag()));
 		}
 		data.put("furnace", items);
-		CompoundTag furnaceTag = furnace.saveWithoutMetadata();
+
+		CompoundTag tag1 = new CompoundTag();
+		furnace.save(tag1);
+
+		CompoundTag furnaceTag = tag1;
```
Comment (Snownee) [inline src/main/java/snownee/jade/addon/vanilla/FurnaceProvider.java]: Unnecessary variable assignment

**Review by Snownee — CHANGES_REQUESTED** [review]

**Comment by Curodenz** [comment]
OK, I've fix them

**Comment by Snownee** [comment]
1.17.1 requires Java 16, which seems to break the debug functionality

**Comment by Snownee** [comment]
Never mind, I have fixed the problem. I am pushing it based on another branch, Thank you.

**Pull request closed without merging**